src/charm.py

```diff
@@ -38,12 +39,13 @@ def __init__(self, *args):
         # From this point forward, our Charm object will only interact with the
         # adapter and not directly with the framework.
         self.fw_adapter = FrameworkAdapter(self.framework)
-
+        self.prometheus = PrometheusInterface(self, 'http-api')
         # Bind event handlers to events
         event_handler_bindings = {
             self.on.start: self.on_start,
             self.on.config_changed: self.on_config_changed,
             self.on.upgrade_charm: self.on_upgrade,
+            self.prometheus.on.new_client: self.prometheus.new_client_handler
```

Comment:
The handler should be placed in the charm code, not the interface since the latter should really only be responsible for passing information about side A to side B or vice versa. So in this case, it should be `self.prometheus.on.new_client: self.on_new_client`

Reply:
On second though, shouldn't this event binding be done in the interface itself rather than in the charm just like the way the interface in the Grafana side does it: https://github.com/relaxdiego/charm-k8s-grafana/blob/master/src/interface_http.py#L93

Sorry, I'm still grappling with the entire idea behind interfaces. Some parts still confuse me.